### DISCUSSION TITLE
Date Field: Timezone differs between admin panel input, and rest response

### DISCUSSION THREAD

**Discussion by quornik** [payloadcms/payload#726] [Q&A] [answered]
# Bug Report

Date provided in Date Field is inconsistant with it's response form provided via Rest API. Propable cause is the timezone difference. 

## Expected Behavior

I provide data via Data Field. The provided date is like 24.05.2022, 10:00 AM. The rest response should be exactly the same. 

## Current Behavior

The date in response is however : 24.05.2022, 8:00 AM - two hours prior to what I've provided.

## Possible Solution

Check the timezone of Admin frontend and adjust the timezone stamp accordingly, or build functionality to adjust timezone of API response via admin. If I understand correctly, at this point the response date is set via server time, which may differ from the timezone of the browser in whicz admin is working.

## Steps to Reproduce
1. Create Date Field in collection.
2. Provide date to the field via admin panel. 
3. Chech the rest response of the field. In my case, the response is different.

## Detailed Description

Current Payload version, Latest Chrome browser, UTC+1 timezone. Payload hostedn on Heroku, without the ability to granular control of the timezone of server.

**Comment by DanRibbens** [comment]
Thanks for reporting this @quornik. We're considering what the configuration settings will be for timezone and what effects it will have for the admin UI and APIs to solve the issue. We'll update again when we have a plan or fix ready.

**Comment by quornik** [comment]
Thanks @DanRibbens.
You should also consider if I haven't rushed to quickly to call it a bug. Maybe it works properly and what it actually needs is just slightly updated documentation to the field. A warning of some kind that one should always localize the response.

If you think about it, the best practice is to **adjust the timezone always on frontend**, after the dateObject has been received and befere it is displayed, because the user's context should determine the way date is interpreted & displayed.

The scenario in which I had a problem with the way it works now, was this particular situation that I did not localize the response date before displaying. In every other situation it works fine.

**Comment by jmikrut** [comment]
> If you think about it, the best practice is to adjust the timezone always on frontend, after the dateObject has been received and befere it is displayed, because the user's context should determine the way date is interpreted & displayed.

Exactly, @quornik. This isn't really a bug _per se_, rather, it's a potentially useful admin UX enhancement.

We're thinking through how to make this happen as Dan said and will report back here once we have more!

**Comment by DanRibbens** [comment, marked as answer]
Since we deterimined there isn't anything in Payload to _fix_ I'm moving this to a discussion for others to see.

**Comment by lucaslcode** [comment]
Here's a use case that doesn't fit the current setup too well - I'm adding events using payload, however the events I'm adding take place in a different timezone to where I am currently.

It seems payload converts my local time to UTC and stores that - it means I have to do some calculating to input the time I want as it would be in my current timezone.

I think a possible solution would be to have the ability to select the timezone of the time you are entering, though I'm not sure if `react-datepicker` supports this.

**Reply by revnelson** [reply]
This is exactly the issue I'm currently facing. Selecting timezone in the date picker would be an amazing additional `pickerAppearance` option. Barring that, even just having an option to save the date in UTC time would cut down on a lot of workarounds. Then we can have a separate selection field that we populate with timezones and group it with the date field so we can use hooks to calculate the correct date for different front-end users.

**Reply by ssyberg** [reply]
+1 this should be a priority! We're having to rip out all of our Date Fields because of this!

**Reply by iamlinkus** [reply]
+1 also!

**Comment by skysarwer** [comment]
Chiming in to say I am experiencing issues with this as well. It would be really nice to have the ability to set a specific timezone for the datefield (or a global config for all date fields) to conform to. 

Maybe there is already a way to do this within the Date field config that isn't clearly documented?

**Reply by paulpopus** [reply]
This is being added in via https://github.com/payloadcms/payload/pull/14410

**Comment by paulpopus** [comment]
Closing this discussion as it's outdated.

We have since shipped timezones support, see: https://payloadcms.com/docs/admin/overview#timezones

Some outstanding requests are also being shipped via https://github.com/payloadcms/payload/pull/14410
